### ISSUE TITLE
final layer

### ISSUE THREAD

**Issue by pyagoubi** [CVxTz/recommender_transformer#5]
Not really an issue, more a question

The final layer takes the embedding size as input and outputs the vocab size
self.linear_out = Linear(channels, self.vocab_size)

But that leaves the sequence untouched. So the final output will have dimensions (batch_size, sequence = 120, vocab_size), so for every token the is vocab_size representation, but I would expect only the sequence as output (batch_size, sequence). Is there a pooling missing?
So I do not really understand how the final output layer works, would really appreciate any clarification if its not a bug.

**Comment by ankitkr3** [comment]
@pyagoubi hi how did you generate the model.cpkt file? can you please help with the process.
Thanks.

**Comment by CVxTz** [comment]
The training happens here: https://github.com/CVxTz/recommender_transformer/blob/main/recommender/training.py 

@pyagoubi  The extra dim of vocab_size is a the list of probabilities on all candidate recommendation.

**Comment by pyagoubi** [comment]
@CVxTz  thanks, but shouldnt then there be a softmax or argmax function?
